### PR TITLE
Delete pipelines

### PR DESCRIPTION
**Pull request by aclevername**

### CODE REVIEW

File: controllers/dynamic_resource_request_controller.go [hunk 1]
```diff
@@ -181,30 +127,42 @@ func (r *dynamicResourceRequestController) Reconcile(ctx context.Context, req ct
...
+		}
+
+		if existingDeletePipelinePod == nil {
+			deletePipelinePod := pipeline.NewDeletePipelinePod(resourceRequest, r.deletePipelines, resourceRequestIdentifier, r.promiseIdentifier)
+			logger.Info("Creating Delete Pipeline for Promise resource request: " + resourceRequestIdentifier + ". The pipeline will now execute...")
+			err = r.Client.Create(ctx, &deletePipelinePod)
+			if err != nil {
+				logger.Error(err, "Error creating Pod")
+				y, _ := yaml.Marshal(&deletePipelinePod)
+				logger.Error(err, string(y))
```
[the remark this answers is not in the document]
Comment (aclevername) [reply controllers/dynamic_resource_request_controller.go:148]: this was a copy and paste of what we do today for the configure pipeline, that doesn't mean we should do it but I left it in.

File: controllers/dynamic_resource_request_controller.go [hunk 2]
```diff
@@ -224,6 +182,70 @@ func (r *dynamicResourceRequestController) deleteResources(ctx context.Context,
...
+	return false
+}
+
+func (r *dynamicResourceRequestController) triggerOrGetDeleteWorkflow(ctx context.Context, resourceRequest *unstructured.Unstructured, finalizer string, logger logr.Logger) (bool, error) {
+	//get pod
+	//if exists check status
+	//otherwise create the pod
+	return false, nil
+}
+
```
Comment (kirederik) [inline controllers/dynamic_resource_request_controller.go:248]: is this used? seems... incomplete?

File: controllers/dynamic_resource_request_controller.go [hunk 3]
```diff
@@ -224,6 +182,70 @@ func (r *dynamicResourceRequestController) deleteResources(ctx context.Context,
...
+	if len(pods) == 0 {
+		return nil, nil
+	}
+	return &pods[0], nil
+}
+
+func (r *dynamicResourceRequestController) configurePipelinePodHasBeenCreated(resourceRequestIdentifier string) bool {
+	pods, err := r.getPodsWithLabels(pipeline.ConfigurePipelineLabels(resourceRequestIdentifier, r.promiseIdentifier))
+	if err != nil {
+		return false
```
Comment (kirederik) [inline controllers/dynamic_resource_request_controller.go:199]: should we assume that an error here means the pod hasn't been created? 🤔

File: lib/pipeline/shared.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	labels["kratix-pipeline-type"] = pipelineType
+	return labels
+}
+
+func pipelineName(pipelineType, promiseIdentifier string) string {
+	return pipelineType + "-pipeline-" + promiseIdentifier + "-" + getShortUuid()
+}
+
+func getShortUuid() string {
+	envUuid, present := os.LookupEnv("TEST_PROMISE_CONTROLLER_POD_IDENTIFIER_UUID")
```
Comment (kirederik) [inline lib/pipeline/shared.go:52]: although this seem to be there before, it's kinda bad to have some code here just to enable our testing 🤔

File: lib/pipeline/shared.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+				Name:      "vol0",
+			},
+		},
+	}
+
+	volume := v1.Volume{Name: "vol0", VolumeSource: v1.VolumeSource{EmptyDir: &v1.EmptyDirVolumeSource{}}}
+	return container, volume
+}
+
+func CommonPipelineLabels(resourceRequestIdentifier, promiseIdentifier string) map[string]string {
```
Comment (kirederik) [inline lib/pipeline/shared.go:34]: `pipeline.CommonPipelineLabels` could read better if this was just `pipeline.CommonLabels`?

File: lib/pipeline/shared.go (hunk 4 above)
Comment (aclevername) [reply lib/pipeline/shared.go:52]: agreed

File: controllers/dynamic_resource_request_controller.go (hunk 2 above)
Comment (aclevername) [reply controllers/dynamic_resource_request_controller.go:248]: oops I this isn't used, will delete

File: controllers/dynamic_resource_request_controller.go (hunk 3 above)
Comment (aclevername) [reply controllers/dynamic_resource_request_controller.go:199]: good call, should probably return an error

File: lib/pipeline/configure.go [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+	"strconv"
+	"strings"
+
+	platformv1alpha1 "github.com/syntasso/kratix/api/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func NewConfigurePipelinePod(rr *unstructured.Unstructured, pipelines []platformv1alpha1.Pipeline, resourceRequestIdentifier, promiseIdentifier string) v1.Pod {
```
Comment (kirederik) [inline lib/pipeline/configure.go:15]: should this be `NewResourceConfigurePipeline`?

File: lib/pipeline/delete.go [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+import (
+	"strconv"
+
+	platformv1alpha1 "github.com/syntasso/kratix/api/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func NewDeletePipelinePod(rr *unstructured.Unstructured, pipelines []platformv1alpha1.Pipeline, resourceRequestIdentifier, promiseIdentifier string) v1.Pod {
```
Comment (kirederik) [inline lib/pipeline/delete.go:12]: should this be `NewResourceDeletePipeline`?

**Pull request merged**